### PR TITLE
Polyhedron demo - Meta reconstruction plugin

### PR DESCRIPTION
**Pull request by sgiraudot**
Replace the 3 reconstruction plugins (poisson, scale space and advancing front) by one single plugin with a unique dialog.

In addition, I added an "automatic" mode that analyses the point cloud and selects the most adapted method with automatically computed parameters. It is very empiric and not guaranteed to work in general so it cannot go into a CGAL package but it gives acceptable results in general and provides an easy access to the reconstruction algorithms of CGAL.

This has not been merged into integration yet.

### CODE REVIEW

**Comment by sloriot** [comment]
Merged in [CGAL-4.8-Ic-29](https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.8-Ic-29.shtml).

**Comment by sloriot** [comment]
Testsuite shows errors on several platforms.

**Comment by sgiraudot** [comment]
Tested in [CGAL-4.8-lc-33](https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.8-Ic-33.shtml). Everything went fine.

File: Polyhedron/demo/Polyhedron/Polyhedron_demo_surface_reconstruction_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,851 @@
...
+  Q_PLUGIN_METADATA(IID "com.geometryfactory.PolyhedronDemo.PluginInterface/1.0")
+
+  Q_INTERFACES(CGAL::Three::Polyhedron_demo_plugin_interface)
+  QAction* actionScaleSpaceReconstruction;
+
+public:
+  void init(QMainWindow* mainWindow, CGAL::Three::Scene_interface* scene_interface) {
+
+    actionScaleSpaceReconstruction = new QAction(tr("Surface surface reconstruction"), mainWindow);
+    actionScaleSpaceReconstruction->setObjectName("actionSurfaceReconstruction");
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/Polyhedron_demo_surface_reconstruction_plugin.cpp]: Surface surface reconstruction? Did you mean 3D surface reconstruction?

Comment (sgiraudot) [inline Polyhedron/demo/Polyhedron/Polyhedron_demo_surface_reconstruction_plugin.cpp]: Actually I just meant "Surface reconstruction" (query replace fail...).

Also I noticed that I kept _actionScaleSpaceReconstruction_ as the name of the variable which is not consistent (should be _actionSurfaceReconstruction_).

I'm correcting these right away.

**Comment by sloriot** [comment]
Only the last commit was not tested but it is trivial and tested locally.
Errors in the testsuite are not related to this plugin.

**Pull request merged**